### DISCUSSION TITLE
Issue on page /user-guide/time-series.html

### DISCUSSION THREAD

**Discussion by gagebeni** [pydata/xarray#5485] [Q&A] [answered]
Hi all
I want to know if it is possible to index on the lat two points having different coordinates like this
    mask_lon = (data_sst.lon> = x1) & (data_sst.lon <= x0)
    mask_lat = (data_sst.lat == y0, y1) & (data_sst.lat == y0, y1)
and then extract the transect resulting from this indexing like this:
    transect = data_sst.SST.where (mask_lon & mask_lat, drop = True)
Thank you in advance.

**Comment by max-sixty** [comment]
Could you please review the issue template re a MVCE?

**Comment by gagebeni** [comment]
i am not yet experimenting enough on python. My problem is that I wrote some code to draw a line using a colon at the beginning and at the end of my line I used numpy exclusively and since it is an xrarray array it takes me a long time. My data is all of dimensions lat, lon, SST, (1501,2001), and here is my code:

        import xarray as xr
        import glob
        from numpy import *
        import numpy as np
        import matplotlib.pyplot as plt
        from scipy.signal import get_window
        import math
        import pandas as pd
        import numpy as np
        
        filepaths_TEMP = "./ 2014 / MUR-SST_2014__Day_027_27-Jan-2014.nc"
        
        data_sst = xr.open_dataset (filepaths_TEMP)
        lon_sst = data_sst.variables ['lon'] [:]
        lat_sst = data_sst.variables ['lat'] [:] #
        sst = data_sst.variables ['SST'] [:]
        #% to draw the line
        latlim = [50, 50.5];
        lonlim = [-10.2];
        n1 = 1200; #% number of point along profile
        delta_lon = -10 - (2);
        delta_lat = 50 - 50.5;
        
        #cd (input_path1)
        #matfiles = dir (fullfile ('*. nc'));
        # %%
        Flat = latlim [1] + np.arange (0, n1) * delta_lat / n1
        Plon = lonlim [1] + np.arange (0, n1) * delta_lon / n1
        transect = np.zeros ((1200,1))
        for ii in range (0, len (Flat)):
              
            lat_S1 = Flat [ii];
            lon_S1 = Plon [ii];
            a_1 = abs (lat_S1-lat_sst);
            a_2 = abs (lon_S1-lon_sst);
            a_3 = (a_1 + a_2);
            m_3 = a_3 [:]. min ()
            [px, py] = np.where (m_3 == a_3);
            px = px [0];
            py = py [0]; # clear a_1 a_2 a_3 m_3
            transect [ii] = sst [px, py];

My question is, isn't there a way to make it simpler and faster, especially thank you very much in advance.

**Comment by dcherian** [comment, marked as answer]
I recommend looking at `.query` and `.isin`. 

`transect = data_sst.SST.where (mask_lon & mask_lat, drop = True)` looks like it could work. Did you try it?

This webpage might also be useful: https://xarray.pydata.org/en/stable/user-guide/indexing.html#more-advanced-indexing
